### PR TITLE
chore(discover-snql): Add some docstrings and restructure a bit

### PR DESCRIPTION
**Pull request by wmak**
- This is mostly just some docstrings on the fields/filter classes
- But also renamed `resolve_field` to `resolve_column` so that we could
  have a `resolve_field` function that is only for fields and so its
  clearer that `resolve_column` resolves regardless of column type
- Also adjusted types on parse_query and related functions cause i
  noticed that it always returns a sequence, regardless of whether query
  is None or empty string

### CODE REVIEW

File: src/sentry/search/events/fields.py [hunk 1]
```diff
@@ -2309,28 +2309,40 @@ def __init__(self, dataset: Dataset, params: ParamsType):
...
-        if self.is_field_alias(field):
+        elif self.is_field_alias(field):
             return self.resolve_field_alias(field)
+        else:
+            return self.resolve_field(field)
 
+    def resolve_field(self, field: str) -> Column:
+        """Given a public discover field, resolve the alias based on the Query's
+        dataset and return the Snql Column
+        """
```
Comment (Zylphrex) [inline src/sentry/search/events/fields.py:2345]: Hm... doesn't a public discover field imply we're working with the discover dataset?

File: src/sentry/search/events/fields.py [hunk 2]
```diff
@@ -2386,19 +2401,25 @@ def resolve_orderby(self, orderby: Optional[Union[List[str], str]]) -> List[Orde
...
         raise InvalidSearchQuery("Cannot order by a field that is not selected.")
 
-    def is_field_alias(self, alias: str) -> bool:
-        return alias in self.field_alias_converter
+    def is_field_alias(self, field: str) -> bool:
+        """Given a public discover field, check if it's a field alias"""
+        return field in self.field_alias_converter
 
     def resolve_field_alias(self, alias: str) -> SelectType:
+        """Given a field alias convert it to its corresponding snql"""
```
Comment (Zylphrex) [inline src/sentry/search/events/fields.py:2409]:
```suggestion
        """Given a field alias, convert it to its corresponding snql"""
```

File: src/sentry/search/events/fields.py [hunk 3]
```diff
@@ -2343,6 +2355,9 @@ def resolve_field(self, field: str) -> SelectType:
             raise InvalidSearchQuery(f"Invalid characters in field {field}")
 
     def resolve_orderby(self, orderby: Optional[Union[List[str], str]]) -> List[OrderBy]:
+        """Given a list of public discover alias optionally prefixed by a `-` to
+        represent direction, construct a list of Snql Orderbys
+        """
```
Comment (Zylphrex) [inline src/sentry/search/events/fields.py:2360]:
```suggestion
        """Given a list of public discover aliases, optionally prefixed by a `-` to
        represent direction, construct a list of Snql Orderbys
        """
```

File: src/sentry/search/events/filter.py [hunk 4]
```diff
@@ -1043,7 +1047,9 @@ def __init__(self, dataset: Dataset, params: ParamsType):
             TEAM_KEY_TRANSACTION_ALIAS: self._key_transaction_filter_converter,
         }
 
-    def parse_query(self, query: Optional[str]) -> Optional[Sequence[SearchFilter]]:
+    def parse_query(self, query: Optional[str]) -> ParsedTerms:
+        """Given a user's query string construct a list of filters that can be
+        then used to construct the conditions of the Query"""
```
Comment (Zylphrex) [inline src/sentry/search/events/filter.py:1052]:
```suggestion
        """Given a user's query, string construct a list of filters that can be
        then used to construct the conditions of the Query"""
```

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/search/events/fields.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/fields.py:2345]: hmmm does `public dataset field` read better?

Comment (wmak) [reply src/sentry/search/events/fields.py]: went with just removing references to `discover` so just `public field` 🤷

**Pull request merged**